plugins/gin/gin.go

```diff
@@ -0,0 +1,65 @@
+// Copyright 2019 Tetrate Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/*
+Package http contains several client/server http plugin which can be used for integration with net/http.
+*/
+
+package gin
+
+import (
+	"fmt"
+	"github.com/gin-gonic/gin"
+	"github.com/tetratelabs/go2sky"
+	"github.com/tetratelabs/go2sky/propagation"
+	"github.com/tetratelabs/go2sky/reporter/grpc/common"
+	"strconv"
+	"time"
+)
+
+const (
+	httpServerComponentID int32 = 49
+)
+
+//Middleware gin middleware return HandlerFunc  with tracing.
+func  Middleware(tracer *go2sky.Tracer) gin.HandlerFunc {
+	return func(c *gin.Context) {
+		if tracer != nil {
```

Comment:
just a nit, what about using early termination to have one identation level less?

Reply:
i think that  if middleware err  will go to next middleware.